---
title: "sequencing QC"
author: "Renee Matthews"
date: "2025-03-13"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
			dev = c("png","pdf")
	
)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```

##### Loading Packages

```{r package loading}
library(tidyverse)
library(readr)
library(kableExtra)
library(DT)
library(readr)
```

### data loading

```{r data loading}

sampleinfo <- read_delim("data/sampleinfo.txt", delim = "\t")

# sampleinfo %>%
#   mutate(other_time = if_else(time=="24+24","24r",other_time))
  # write_delim(sampleinfo,"data/sampleinfo.txt", delim = "\t")

multiqc_gene_stats_notrim <- read_delim("data/multiqc_genral_stats_notrim.txt",delim = "\t")
multiqc_gene_stats_trim <- read_delim("data/multiqc_genral_stats_trim.txt",delim = "\t")

multiqc_fastqc_notrim <- read_delim("data/multiqc_fastqc_notrim.txt",delim = "\t")
multiqc_fastqc_trim <- read_delim("data/multiqc_fastqc_trim.txt",delim = "\t")
    
```

Combining the dataframes for visualization

```{r datacombine}
combo_notrim_df <- multiqc_fastqc_notrim %>% 
 extract(., Sample, into = c("prefix","read"), regex= "(.+)_R(\\d+)", remove=FALSE) %>% 
  mutate(read = paste0("R", read)) %>% 
  left_join(., sampleinfo, by =c("prefix"="library_ID")) %>% 
  left_join(., multiqc_gene_stats_notrim, by = c("Sample" = "Sample")) %>% 
  mutate(ind=factor(ind, levels = c("1","2","3","4","5"))) %>% 
  mutate(trt=factor(trt, levels = c("VEH","5FU","DOX"))) %>% 
  mutate(other_time=factor(other_time, levels=c("24t","24r","144r")))

combo_trim_df <- multiqc_fastqc_trim %>% 
 extract(., Sample, into = c("prefix","read"), regex= "(.+)_R(\\d+)", remove=FALSE) %>% 
  mutate(read = paste0("R", read)) %>% 
  left_join(., sampleinfo, by =c("prefix"="library_ID")) %>% 
  left_join(., multiqc_gene_stats_trim, by = c("Sample" = "Sample")) %>% 
  mutate(ind=factor(ind, levels = c("1","2","3","4","5"))) %>% 
  mutate(trt=factor(trt, levels = c("VEH","5FU","DOX"))) %>% 
  mutate(other_time=factor(other_time, levels=c("24t","24r","144r")))
```

### Basic Stats

```{r Basic stats}
combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
  group_by(trt,other_time,histone_mark) %>% 
  tally() %>% 
  ggplot(., aes(x = other_time, y= n))+
  geom_col(position="dodge",aes(fill=trt)) + 
  facet_wrap(~histone_mark)+
  theme(axis.text.x=element_text(angle=90))+
  ylab("number of samples")+
  ggtitle("Breakdown of samples by mark and trt-time")

combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
  group_by(trt,other_time,histone_mark) %>% 
  tally() %>% 
  pivot_wider(., id_cols=c(trt,other_time), names_from = histone_mark, values_from = n) %>% 
  kable(.,caption = ("Sample counts")) %>% 
  kable_paper("striped", full_width = FALSE) %>%
  kable_styling(full_width = FALSE,font_size = 16) %>%
  scroll_box(width = "100%", height = "500px")

```


### Visualization of counts

```{r counts viz, fig.height=8, fig.width=12}

combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = Sample, y= `Total Sequences`))+
  geom_col(aes(fill=histone_mark)) + 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  ylab("sequene count")+
  ggtitle("Read counts by sample and histone mark")+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))

combo_trim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = Sample, y= `Total Sequences`))+
  geom_col(aes(fill=histone_mark)) + 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  ylab("sequene count")+
  ggtitle("Read counts by sample and histone mark trimmed adapters")+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))



combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
   ggplot(., aes(x = histone_mark, y= `Total Sequences`))+
geom_boxplot(aes(fill=histone_mark)) + 
    geom_point(aes(color=ind))+
  facet_wrap(trt~other_time)+
  ylab("count")+
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  ggtitle("Sequence count by trt and time no trim")

combo_trim_df %>% 
  dplyr::filter(read=="R1") %>% 
   ggplot(., aes(x = histone_mark, y= `Total Sequences`))+
geom_boxplot(aes(fill=histone_mark)) + 
    geom_point(aes(color=ind))+
  facet_wrap(trt~other_time)+
  ylab("count")+
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  ggtitle("Sequence count by trt and time trimmed adapter")


```

### Trim information

```{r trim info1}
combo_trim_df %>% 
   ggplot(., aes(x = read, y= avg_sequence_length))+
geom_boxplot(aes(fill=read))

combo_trim_df %>% 
  ggplot(., aes(x = read, y= avg_sequence_length))+
geom_boxplot(aes(fill=histone_mark)) +
  ggtitle("Boxplot of trim read length across histone marks")

combo_trim_df %>% 
  datatable(., options = list(scrollX = TRUE, 
                              scrollY = "400px",
                              scrollCollapse = TRUE,
                              fixedColumns = list(leftColumns =2),
                              fixedHeader= TRUE),
            extensions = c("FixedColumns","Scroller"),
            class = "display")

# combo_trim_df %>% 
#   dplyr::select(prefix,histone_mark,sample_ID:other_time,`Total Sequences`) %>% 
#   distinct() %>% 
#   mutate(File_name=paste0(prefix,".bed"))


```


```{r trim info2,fig.height=8, fig.width=12}
combo_trim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = Sample, y= avg_sequence_length))+
geom_col(aes(fill=histone_mark)) +
  geom_hline( yintercept = 75)+
  theme_classic()+
  ggtitle("Graph of average read length across R1 samples")+ theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))

combo_trim_df %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x = Sample, y= avg_sequence_length))+
geom_col(aes(fill=histone_mark)) +
  geom_hline( yintercept = 75)+
  theme_classic()+
  ggtitle("Graph of average read length across R2 samples")+ theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
  


```

percent gc

```{r GCper,fig.height=6, fig.width=12}
combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = Sample, y= `%GC`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of %GC for R1 not-trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))


combo_trim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = Sample, y= `%GC`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of %GC for R1 trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))

combo_notrim_df %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x = Sample, y= `%GC`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of %GC for R2 not-trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))


combo_trim_df %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x = Sample, y= `%GC`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of %GC for R2 trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))


```


Duplication information
```{r dup viz, fig.height=8, fig.width=8}
combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = histone_mark, y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_boxplot(aes(fill=histone_mark)) + 
    geom_point(aes(color=ind))+
  facet_wrap(trt~other_time)+
  ylab("percent duplication")+
  theme(axis.text.x=element_text(angle=90))+
   ggtitle("Duplication percentage (R1 no trim)")

combo_trim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = histone_mark, y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_boxplot(aes(fill=histone_mark)) + 
    geom_point(aes(color=ind))+
  facet_wrap(trt~other_time)+
  ylab("percent duplication")+
  theme(axis.text.x=element_text(angle=90))+
  ggtitle("Duplication percentage (R1 trimmed)")
  


combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = interaction(other_time,trt), y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_boxplot(aes(fill=histone_mark)) + 
  geom_point(aes(color=ind))+
facet_wrap(~histone_mark)+
  ylab("percent duplication")+
  theme(axis.text.x=element_text(angle=90))+
  ggtitle("Duplication percentage (R1 no trim)")

combo_trim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = interaction(other_time,trt), y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_boxplot(aes(fill=histone_mark)) + 
  geom_point(aes(color=ind))+
facet_wrap(~histone_mark)+
  ylab("percent duplication")+
  theme(axis.text.x=element_text(angle=90))+
  ggtitle("Duplication percentage (R1 trimmed)")


  
```





```{r,Dup all samp,fig.height=6, fig.width=12}
combo_notrim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = Sample, y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of percent duplicates for R1 not-trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous(limits = c(0,100), expand = expansion(mult = c(0, .1)))


combo_trim_df %>% 
  dplyr::filter(read=="R1") %>% 
  ggplot(., aes(x = Sample, y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of percent duplicates for R1 trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( limits = c(0,100),expand = expansion(mult = c(0, .1)))

combo_notrim_df %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x = Sample, y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of percent duplicates for R2 not-trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous(limits = c(0,100), expand = expansion(mult = c(0, .1)))


combo_trim_df %>% 
  dplyr::filter(read=="R2") %>% 
  ggplot(., aes(x = Sample, y= `FastQC_mqc-generalstats-fastqc-percent_duplicates`))+
  geom_col(aes(fill=histone_mark)) +
  theme_classic()+
  ggtitle("Graph of percent duplicates for R2 trimmed")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous(limits = c(0,100), expand = expansion(mult = c(0, .1)))


```

```{r code chunk, echo=TRUE, eval=FALSE}
### creating the sample info file from bowtie2 metrics from the first run:

### import files

metric_1 <-  read_delim("C:/Users/reneels/Documents/Ward Lab/5FU_project/Sample_info/first_run_metrics/metric_1.txt", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE)
metric_2 <- read_delim("C:/Users/renee/Documents/Ward Lab/5FU_project/Sample_info/first_run_metrics/Second_set_info.txt", delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
pull_name_file <- read_csv("C:/Users/renee/Documents/Ward Lab/5FU_project/Sample_info/first_run_metrics/samplesheet.txt", col_names = FALSE)
pull <- pull_name_file[65:94,]

 fun <- cbind(pull, metric_2[,1:8])
 colnames(fun) <- c("X1","X2","X3","X4","X5","X6","X7","X8","X9")
 
 # almost_fun <- cbind(pull_name_file[1:65,],metric_1)
 
 ##remove 229 from almost_fun
# subset_almost_fun <- almost_fun[1:64,]

alignment_specs <- rbind(metric_1,fun)

colnames(alignment_specs) <- c("sample_ID",
                "Paired_Reads",
                "aligned_pairs",
                "aligned_concordant_0",
                "zero",
                "aligned_concordant_1",
                "one",
                "aligned_concordant_g1",
                "percent_alignment",
                "N_A")
colnames(fun) <- c("sample_ID",
                "Paired_Reads",
                "aligned_pairs",
                "aligned_concordant_0",
                "zero",
                "aligned_concordant_1",
                "one",
                "aligned_concordant_g1",
                "percent_alignment",
                "N_A")
saveRDS(fun, "data/first_run_alignment_229andover.RDS")
saveRDS(alignment_specs, "data/first_run_alignment_specs.RDS")

```
```{r alignmentspec}

alignment_temp<-readRDS("data/first_run_alignment_specs.RDS")
for_plots <- alignment_temp %>% 
  left_join(.,sampleinfo, by=c("sample_ID"="library_ID"))%>% 
  dplyr::select(sample_ID:aligned_concordant_0,aligned_concordant_1, aligned_concordant_g1,percent_alignment, histone_mark,ind, trt,other_time) %>% 
  distinct()
```

alignment pair numbers
```{r,fig.height=6, fig.width=12}
for_plots %>% 
  group_by(histone_mark) %>% 
  ggplot(., aes(x=sample_ID, y=Paired_Reads))+
    geom_col(aes(fill = histone_mark)) +
  theme_classic()+
  ggtitle("Graph of total paired reads")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))

for_plots %>% 
  mutate(percent_alignment=parse_number(percent_alignment)) %>% 
  group_by(histone_mark) %>% 
  ggplot(., aes(x=sample_ID, y=percent_alignment))+
    geom_col(aes(fill = histone_mark)) +
  theme_classic()+
  ggtitle("Percent aligned of paired reads") + 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous(expand = expansion(mult = c(0, .1)))+
  coord_cartesian(ylim=c(80,100))


for_plots %>% 
  group_by(histone_mark) %>% 
   mutate(percent_aligned_unique = round(aligned_concordant_1 / Paired_Reads * 100, 2)) %>% 
  ggplot(., aes(x=sample_ID, y=percent_aligned_unique))+
    geom_col(aes(fill = histone_mark)) +
  theme_classic()+
  ggtitle("Percentage of paired reads aligning once")+ 
  theme(axis.text.x=element_text(vjust = .2,angle=90))+
  scale_y_continuous( expand = expansion(mult = c(0, .1)))
```
### Fragment analysis
```{r}
file_list <- list.files(path="C:/Users/renee/Documents/5FU_files/metric_files",
           pattern ="frag_len_count\\.txt$",full.names = TRUE)
read_and_label <- function(file) {
  df <- read_delim(file, delim = "\t", col_names = c("Col1", "Col2"))  # Adjust delimiter if needed
  df <- df %>%
    mutate(File = basename(file))  # Add filename column
  return(df)
}
combined_df <- map_df(file_list, read_and_label)

annotated_combo_df <- combined_df %>% 
  mutate(sample = gsub("_frag_len_count.txt","",File)) %>% 
  left_join(., sampleinfo, by = c("sample"="library_ID"))
```

```{r, fig.width=15}
annotated_combo_df %>% 
  dplyr::filter(histone_mark=="H3K9me3") %>% 
  ggplot(., aes(x=Col1, y=Col2, color = sample))+
  geom_line(size=1)+
  # geom_point(size=2,alpha =0.7)+
  scale_x_continuous(breaks = seq(0, max(annotated_combo_df$Col1), by = 50))+
  labs(title = "Fragment length, H3K9me3",
       x = "Fragment Length (bp)",
       y = "Count",
       color= "Sample")+
  theme_minimal()

annotated_combo_df %>% 
  dplyr::filter(histone_mark=="H3K27me3") %>% 
  ggplot(., aes(x=Col1, y=Col2, color = sample))+
  geom_line(size=1)+
  # geom_point(size=2,alpha =0.7)+
  scale_x_continuous(breaks = seq(0, max(annotated_combo_df$Col1), by = 50))+
  labs(title = "Fragment length, H3K27me3",
       x = "Fragment Length (bp)",
       y = "Count",
       color= "Sample")+
  theme_minimal()

annotated_combo_df %>% 
  dplyr::filter(histone_mark=="H3K27ac") %>% 
  ggplot(., aes(x=Col1, y=Col2, color = sample))+
  geom_line(size=1)+
  # geom_point(size=2,alpha =0.7)+
  scale_x_continuous(breaks = seq(0, max(annotated_combo_df$Col1), by = 50))+
  labs(title = "Fragment length, H3K27ac",
       x = "Fragment Length (bp)",
       y = "Count",
       color= "Sample")+
  theme_minimal()

annotated_combo_df %>% 
  dplyr::filter(histone_mark=="H3K36me3") %>% 
  ggplot(., aes(x=Col1, y=Col2, color = sample))+
  geom_line(size=1)+
  # geom_point(size=2,alpha =0.7)+
  scale_x_continuous(breaks = seq(0, max(annotated_combo_df$Col1), by = 50))+
  labs(title = "Fragment length, H3K36me3",
       x = "Fragment Length (bp)",
       y = "Count",
       color= "Sample")+
  theme_minimal()
```

